tuning/sft_trainer.py

```diff
@@ -360,7 +361,24 @@ def train(
         for x in framework.get_callbacks_and_ready_for_train(model, accelerator):
             trainer.add_callback(x)
 
-    trainer.train()
+    resume_from_checkpoint = None
+    # Check if resume flag is not passed (None), or if flag is true and
+    # output_dir has checkpoints then get last checkpoint from output_dir
+    if (
+        training_args.resume_from_checkpoint is None
+        or training_args.resume_from_checkpoint.lower() == "true"
+    ):
+        resume_from_checkpoint = get_last_checkpoint(training_args.output_dir)
+    else:
+        # `training_args.resume_from_checkpoint` gives string values
+        # Check if flag is false OR flag has checkpoint value for resuming tuning
+        resume_from_checkpoint = (
+            training_args.resume_from_checkpoint
+            if training_args.resume_from_checkpoint.lower() != "false"
+            else False
+        )
```

Comment:
Why not simplify this logic even further and move it into __post_init__ of training_args. 

We can check if `resume_from_checkpoint` is `None` or `True`, and if so, we check if `get_last_checkpoint(training_args.output_dir)` returns a non empty string. If it returns a non empty string, then we set `resume_from_checkpoint` to `True`, else we set `resume_from_checkpoint` to `False`.  In the current condition added, if `get_last_checkpoint` does not return a checkpoint, doesn't it get set to empty string, which is not what we want, isn't it?

We can leave every other case untouched, Isn't it? HF will already handle the other cases well, or is it not handling when `resume_from_checkpoint` is a string `false`?

Reply:
>if `get_last_checkpoint` does not return a checkpoint, doesn't it get set to empty string, which is not what we want, isn't it?

`get_last_checkpoint` either returns a checkpoint string or None.

Reply:
> is it not handling when `resume_from_checkpoint` is a string false ?

When `resume_from_checkpoint` is a string false, `trainer()` needs argument `resume_from_checkpoint` value as boolean.

`trainer()` accepts argument `resume_from_checkpoint` value as boolean True/False, None (none_type), or string with path of checkpoint.

Reply:
So we are adding support for string `true` and `false` additionally, even though HF Trainer does not support it?

Reply:
What is the behaviour if we set the Training argument `resume_from_checkpoint` to be `True` or `False`? Isn't it handling it as a boolean `True` or `False` and behave accordingly?

Reply:
Training argument `resume_from_checkpoint` takes in the path to a folder with a valid checkpoint for the model, as mentioned [here](https://github.com/huggingface/transformers/blob/7591ca5bc583aadbdfe6b2fb36a12c874233323c/src/transformers/training_args.py#L665), and doesn't work with `boolean/String` value `True/False`

Reply:
Then lets go ahead with this change. LGTM. We can refine later if required.